### PR TITLE
Segmented Config changes

### PR DESCRIPTION
**Pull request by chvamshi-xilinx**
#### Problem solved by the commit
We were relying on pr_isolation address to program the PDI earlier, With segmented config support, we need to relax that check. We will get information from new bitmap entry
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
None
#### How problem was solved, alternative solutions (if any) and why they were rejected
Relaxed all the pr_isolation checks in zocl
#### Risks (if any) associated the changes in the commit
None
#### What has been tested and how, request additional testing if necessary
Verified zynqmp/versal aie/non-aie cases
#### Documentation impact (if any)
None

### CODE REVIEW

File: src/runtime_src/core/edge/user/shim.cpp [hunk 1]
```diff
@@ -688,6 +688,8 @@ xclLoadAxlf(const axlf *buffer)
   auto is_flat_enabled = xrt_core::config::get_enable_flat(); //default value is false
   auto force_program = xrt_core::config::get_force_program_xclbin(); //default value is false
   auto overlay_header = xclbin::get_axlf_section(buffer, axlf_section_kind::OVERLAY);
+  //force load PDI if below action mask is set
+  is_pr_platform = (buffer->m_header.m_actionMask & AM_LOAD_PDI) ? true : false;
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/shim.cpp:692]: huh?  Why not change line 687?

File: src/runtime_src/core/edge/user/shim.cpp [hunk 2]
```diff
@@ -1181,6 +1183,8 @@ int shim::prepare_hw_axlf(const axlf *buffer, struct drm_zocl_axlf *axlf_obj)
...
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/shim.cpp:1187]: ditto?

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/edge/user/shim.cpp (hunk 1 above)
Comment (chvamshi-xilinx) [reply src/runtime_src/core/edge/user/shim.cpp]: Thanks @stsoe . Fixed it. It was a bug indeed :-(

File: src/runtime_src/core/edge/user/shim.cpp (hunk 2 above)
Comment (chvamshi-xilinx) [reply src/runtime_src/core/edge/user/shim.cpp]: Fixed.

File: src/runtime_src/core/edge/user/shim.cpp [hunk 3]
```diff
@@ -684,7 +684,7 @@ xclLoadAxlf(const axlf *buffer)
   std::string dtbo_path("");
 
 #ifndef __HWEM__
-  auto is_pr_platform = (buffer->m_header.m_mode == XCLBIN_PR ) ? true : false;
+  auto is_pr_platform = (buffer->m_header.m_mode == XCLBIN_PR || buffer->m_header.m_actionMask & AM_LOAD_PDI) ? true : false;
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/shim.cpp:687]: Kind of strange to do `if (true) true; else false`;  :-)  Maybe drop the ternary?

Comment (chvamshi-xilinx) [reply src/runtime_src/core/edge/user/shim.cpp]: yeah.. Fixed it Soren. Thank you.

**Pull request merged**